### PR TITLE
[BEAM-14323] Improve IDE integration of Spark cross version builds

### PR DESCRIPTION
**Pull request by mosche**
With the current build setup, developer experience is fairly poor when working with the cross version build for Spark (but also similarly for Flink):

- Sources for version specific overrides are copied to a new location and defined as Gradle sources at that location. 
  1) First of all, this is totally unnecessary. These sources are not shared and should be used in place. 
  2) Much more troublesome, the actual sources won't be resolved / checked by any IDE anymore and can't be properly worked on that way. Sadly for no reason at all ...
- The actual shared resources on the other hand are referenced (added to srcDirs) in place. The IDE will randomly assign them to one Spark version module. Typically, for IntelliJ at least, that's the first (lower) one and not the one developers are actively working on.

The suggested changes in this PR are:
- Don't copy version specific overrides
- Only copy shared sources conditionally based on a flag. This allows developers to disable copying to pick a primary version they intend to work on.

Note: This is primary a cosmetic flag to improve IDE integration and has no impact on builds, even if all modules disable copying.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by mosche** [comment]
R: @jbonofre 
R: @echauchot 
R: @aromanenko-dev

**Comment by mosche** [comment]
R: @lukecwik

**Comment by aromanenko-dev** [comment]
Run Spark ValidatesRunner

File: runners/spark/2/build.gradle [hunk 1]
```diff
@@ -22,12 +22,7 @@ project.ext {
   // Set the version of all Spark-related dependencies here.
   spark_version = '2.4.8'
   spark_scala_version = '2.11'
-
-  // Version specific code overrides.
-  main_source_overrides = ['./src/main/java']
-  test_source_overrides = ['./src/test/java']
-  main_resources_overrides = []
-  test_resources_overrides = []
+  copySourceBase = true // enabled to use Spark 3 as primary dev version
```
Comment (aromanenko-dev) [inline runners/spark/2/build.gradle:25]: Is it a typo here? Should it be `Spark2`?

**Review by aromanenko-dev — COMMENTED** [review]
LGTM if it doesn't break a build/testing process

**Comment by aromanenko-dev** [comment]
CC: @ibzib

File: runners/spark/2/build.gradle (hunk 1 above)
Comment (mosche) [reply runners/spark/2/build.gradle:25]: nup, Spark 3 should be the primary version for development ... so we copy sources for Spark 2

File: runners/spark/spark_runner.gradle [hunk 2]
```diff
@@ -49,67 +49,46 @@ configurations {
 }
 
 def hadoopVersions = [
-    "285": "2.8.5",
```
[the remark this answers is not in the document]
Comment (mosche) [reply runners/spark/spark_runner.gradle:52]: auto code formatting

File: runners/spark/2/build.gradle (hunk 1 above)
Comment (aromanenko-dev) [reply runners/spark/2/build.gradle:25]: Maybe choose another name for that then? It sounds a bit ambiguous now for me.

Comment (mosche) [reply runners/spark/2/build.gradle]: 👍 I've rephrased the comment

**Pull request merged**